mirror-url.go

```diff
@@ -119,57 +119,51 @@ func deltaSourceTargets(sourceURL string, targetURL string, isForce bool, isFake
 		return
 	}
 
-	// Setup object difference.
-	objectDifferenceTarget := objectDifferenceFactory(targetClnt)
-
-	// Set default values for listing.
-	isRecursive := true   // recursive is always true for diff.
-	isIncomplete := false // we will not compare any incomplete objects.
-
-	// List all the sources, compare using 'objectDifference' function.
-	for sourceContent := range sourceClient.List(isRecursive, isIncomplete) {
-		if sourceContent.Err != nil {
-			mirrorURLsCh <- mirrorURLs{
-				Error: sourceContent.Err.Trace(sourceClient.GetURL().String()),
-			}
-			continue
-		}
-		if sourceContent.Type.IsDir() {
-			continue
-		}
-		sourceSuffix := strings.TrimPrefix(sourceContent.URL.String(), sourceURL)
-		differ, err := objectDifferenceTarget(targetURL, sourceSuffix, sourceContent.Type, sourceContent.Size, sourceContent.Time)
-		if err != nil {
-			mirrorURLsCh <- mirrorURLs{Error: err.Trace(sourceContent.URL.String())}
-			continue
-		}
-		if differ == differInNone {
+	// List both source and target, compare and return values through channel.
+	for diffMsg := range objectDifference(sourceClnt, targetClnt, sourceURL, targetURL) {
+		if diffMsg.Diff == differInNone {
 			// No difference, continue.
 			continue
-		}
-		if differ == differInType {
-			mirrorURLsCh <- mirrorURLs{Error: errInvalidTarget(sourceSuffix)}
+		} else if diffMsg.Diff == differInType {
+			mirrorURLsCh <- mirrorURLs{Error: errInvalidTarget(diffMsg.SecondURL)}
 			continue
-		}
-		if differ == differInSize && !isForce && !isFake {
-			// Size differs and force not set
-			mirrorURLsCh <- mirrorURLs{Error: errOverWriteNotAllowed(sourceContent.URL.String())}
+		} else if diffMsg.Diff == differInSize {
+			if !isForce && !isFake {
```

Comment:
> Not able to guess why !isFake is needed here ?

Because if fake is set it is still going to show that it is going to copy the contents.